sfyra/pkg/tests/server_class.go

```diff
@@ -234,6 +235,35 @@ func TestServerClassPatch(ctx context.Context, metalClient client.Client, cluste
 		configProvider, err := configloader.NewFromBytes(metadataBytes)
 		require.NoError(t, err)
 
+		// try deleting dummy server before deleting the cluster
+		// it shouldn't break the cluster
+		require.NoError(t, metalClient.Delete(ctx, &dummyServer))
+
+		serverStillExists := true
+		labelSelector, err := labels.Parse("dummy-server=")
+		require.NoError(t, err)
+
+		dummyServers := &v1alpha1.ServerList{}
+
+		// try checking that server is still there 10 times
+		retry.Constant(10*time.Second, retry.WithUnits(1*time.Second)).Retry(func() error { // nolint:errcheck
```

Comment:
nit: I guess this might be simplified to check server existence in 10 seconds? we don't really need to use retry here, or we could do a simple for loop asserting that object exists on each iteration?

Plus we could use `.Get()` method instead of `List()` to get object by name directly and check that there're no errors.